services/server/src/common/errors/GenericErrorHandler.ts

```diff
@@ -9,7 +10,10 @@ export default function genericErrorHandler(
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   _next: any,
 ): void {
-  const errorCode = +err.code || err.status || 500;
+  const errorCode = +err.statusCode || err.status || 500;
+  if (errorCode === 500) {
+    logger.error("Unexpected server error", { error: err });
+  }
```

Comment:
If we add this logger here, then we have to remove the log from any other place in which we throw I suppose?

Reply:
I moved it mainly here, because we had a log in the `safeHandler` which is removed now. So before removing it, I think there were already duplicate logs. 

Imo, the duplicate logs are no problem, because in the best case we should never reach this point. It should rather be a way to remind us that we have to handle something differently or fix something.